Python/pylifecycle.c

```diff
@@ -1644,7 +1644,9 @@ Py_FinalizeEx(void)
 
     /* Get current thread state and interpreter pointer */
     PyThreadState *tstate = _PyRuntimeState_GetThreadState(runtime);
+#if (defined(Py_REF_DEBUG) || defined(Py_TRACE_REFS) || defined(WITH_PYMALLOC))
```

Comment:
Could you elaborate why this guards here? This is the only one I don't understand clearly.

Reply:
See lines 1671 to 1679. The ``interp`` variable is only used when one of the macros is defined.

https://github.com/tiran/cpython/blob/bpo-40998-ubsan-warnings/Python/pylifecycle.c#L1671-L1679

Reply:
Now that you mention it, I could simplify the code by replacing ``interp`` with ``tstate->interp``.